File: 6_kyu/Counting_Duplicates.kt
fun duplicateCount(text: String): Int {
    val counter = HashMap<Char, Int>(36)
    for (c in text) {
        val C = c.toUpperCase()
        counter[C] = counter.getOrDefault(C, 0) + 1
    }
    return counter.count { it.value > 1 }
}

File: 6_kyu/Counting_Duplicates.r
# Count the number of characters more than 1-time in the text.
# Handels characters case insensitive.
duplicate_count <- function(text){
  # Convert all letters to lower case to ignore the case in the input text.
  # Convert to characters from the (multibyte) string.
  char_data <- strsplit(tolower(text), NULL, fixed=TRUE)
  # Count the characters.
  char_frequency <- table(char_data)
  # Return number of factors with more than one occurrence.
  sum(char_frequency>1)
}
